src/Utilities/ErrorHandling/Error.hpp

```diff
@@ -34,48 +35,33 @@
 // 20160415) can't figure out that the else branch and everything
 // after it is unreachable, causing warnings (and possibly suboptimal
 // code generation).
-#define ERROR(m)                                                            \
-  do {                                                                      \
-    disable_floating_point_exceptions();                                    \
-    std::ostringstream avoid_name_collisions_ERROR;                         \
-    /* clang-tidy: macro arg in parentheses */                              \
-    avoid_name_collisions_ERROR << m; /* NOLINT */                          \
-    abort_with_error_message(__FILE__, __LINE__,                            \
-                             static_cast<const char*>(__PRETTY_FUNCTION__), \
-                             avoid_name_collisions_ERROR.str());            \
-  } while (false)
-
-/*!
- * \ingroup ErrorHandlingGroup
- * \brief prints an error message to the standard error and aborts the
- * program.
- *
- * CERROR is just like ERROR and so the same guidelines apply. However, because
- * it does not use std::stringstream it can be used in some constexpr
- * functions where ERROR cannot be.
- * \param m error message as a string, may need to use string literals
- */
-#define CERROR(m)                                                             \
+#define ERROR(m)                                                              \
   do {                                                                        \
-    breakpoint();                                                             \
-    sys::abort("\n################ ERROR ################\nLine: "s +         \
-               std::to_string(__LINE__) + " of file '"s + __FILE__ + "'\n"s + \
-               m + /* NOLINT */                                               \
-               "\n#######################################\n"s);               \
+    if (cpp20::is_constant_evaluated()) {                                     \
```

Comment:
Question. Should this be

```cpp
if constexpr (cpp20::is_constant_evalutated())
```

if `cpp20::is_constant_evalutated()` is a `constexpr bool`? Or does that also create longer compile times? Just curious.

Reply:
Ahh ok yes that makes sense, thank you for the thorough explanation!

Reply:
You're welcome! I had this debate with myself yesterday, as well